terminus-ui/tabs/src/collection/tab-collection.component.ts

```diff
@@ -0,0 +1,438 @@
+import {
+  AfterContentChecked,
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import { untilComponentDestroyed } from '@terminus/ngx-tools';
+import { coerceNumberProperty } from '@terminus/ngx-tools/coercion';
+import { TsStyleThemeTypes } from '@terminus/ui/utilities';
+import { merge, Subscription } from 'rxjs';
+
+import { TsTabHeaderComponent } from './../header/tab-header.component';
+import { TsTabComponent } from './../tab/tab.component';
+
+
+/**
+ * A change event emitted on focus or selection changes
+ */
+export class TsTabChangeEvent {
+  constructor(
+    public index: number,
+    public tab: TsTabComponent | null,
+  ) {}
+}
+
+/**
+ * Possible horizontal layout alignment choices for header tabs
+ */
+export type TsTabAlignmentOptions
+  = 'start'
+  | 'center'
+  | 'end'
+  | 'stretch'
+;
+
+/**
+ * Possible positions for the tab header
+ */
+export type TsTabHeaderPosition
+  = 'above'
+  | 'below'
+;
+
+// Unique ID for each instance
+let nextUniqueId = 0;
+
+
+/**
+ * A collection of {@link TsTabComponent}s
+ *
+ * @example
+ * <ts-tab-collection
+ *               headerPosition="above"
+ *               [selectedIndex]="2"
+ *               tabAlignment="start"
+ *               theme="primary"
+ *               (animationFinished)="myFunc()"
+ *               (focusChange)="myFunc($event)"
+ *               (selectedIndexChange)="myFunc($event)"
+ *               (selectedTabChange)="myFunc($event)"
+ * >
+ *              <ts-tab label="First">
+ *                Content 1
+ *              </ts-tab>
+ *
+ *              <ts-tab label="Second">
+ *                Content 2
+ *              </ts-tab>
+ * </ts-tab-collection>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-master/components/tabs</example-url>
+ */
+@Component({
+  selector: 'ts-tab-collection',
+  templateUrl: './tab-collection.component.html',
+  styleUrls: ['./tab-collection.component.scss'],
+  host: {
+    class: 'ts-tab-collection',
+    '[class.ts-tab-collection--inverted-header]': 'headerPosition === "below"',
```

Comment:
lines 89-98
We are not testing the DOM in search of these class names

Reply:
We are in the collection spec. Most of our real tests are there